### PR TITLE
COBRA-2944 - added pipeline status checks

### PR DESCRIPTION
**Pull request by trevorlinton**

### CODE REVIEW

File: src/components/Releases/Logs.jsx [hunk 1]
```diff
@@ -82,7 +82,7 @@ export default class Logs extends Component {
       );
     }
     return (
-      <pre style={style.logs}><Ansi>{this.state.logs}</Ansi></pre>
```
Comment (trevorlinton) [inline src/components/Releases/Logs.jsx]: This had to be removed as it was causing stack range errors on large amount of logs (budgetapi-adm-dev in Maru is a good example).

File: src/config/GlobalStyles.jsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
+
```
Comment (trevorlinton) [inline src/config/GlobalStyles.jsx]: I created this to help centralize the styles I was using that I couldn't find a way of implementing with Material UI. 

Hopefully it'll help maintain the framework as well as make it easier to update it if we find a solution within Material UI. For example, the "Danger Button" style may end up being introduced into a new version of material ui, in the meantime the styles for it are here and we can search by the key reference (`DangerButton`) if we want to stop using a custom style and start using material ui instead.

File: src/services/api/index.js [hunk 3]
```diff
@@ -228,14 +228,19 @@ function redoBuild(app, build) {
...
 
 async function getReleases(app) {
-  return Promise.all((await axios.get(`/api/apps/${app}/releases`))
-    .data
-    .slice(-10)
-    .map(async release => Object.assign(release, { slug: (await axios.get(`/api/slugs/${release.slug.id}`)).data })));
+  return axios.get(`/api/apps/${app}/releases`)
 }
 
-async function getRawReleases(app) {
```
Comment (trevorlinton) [inline src/services/api/index.js]: I just standardized getReleases to just fetch the releases and moved the slice/map statements that used to be here into the component fetching the data.  I then changed anything using getRawReleases to just use getReleases.

File: src/services/util/index.js [hunk 4]
```diff
@@ -114,7 +114,49 @@ function clearHistory() {
   localStorage.removeItem('akkeris_history');
 }
 
+function getDateDiff(date /* : Date */) {
```
Comment (trevorlinton) [inline src/services/util/index.js]: I needed this other places, so I just included it in the util/index.js and pulled it out of the Releases/Activity tab.

File: src/components/AutoSuggest.jsx [hunk 5]
```diff
@@ -8,11 +8,13 @@ import PropTypes from 'prop-types';
...
-    width: '300px',
+    width:'100%',
+    minWidth: '300px',
   },
   container: {
+    width:'100%',
     position: 'relative',
-    width: '300px',
+    minWidth: '300px',
   },
```
Comment (trevorlinton) [inline src/components/AutoSuggest.jsx]: I needed the auto suggest to fill the width of the container, I couldn't find away to pass this in as a prop but from what I can see with other places its used it didn't have much of an affect.

File: src/components/AutoSuggest.jsx [hunk 6]
```diff
@@ -167,7 +169,7 @@ class AutoSuggest extends Component {
 
 
   render() {
-    const { classes, errorText, className, placeholder } = this.props;
+    const { classes, errorText, labelText, className, placeholder } = this.props;
     const autoSuggestProps = {
```
Comment (trevorlinton) [inline src/components/AutoSuggest.jsx]: I needed the ability to place a label on the autosuggest, its optional however.

File: src/components/Pipelines/NewPipelineCoupling.jsx [hunk 7]
```diff
@@ -1,147 +0,0 @@
-import React, { Component } from 'react';
```
Comment (trevorlinton) [inline src/components/Pipelines/NewPipelineCoupling.jsx]: This was replaced with CreateOrUpdatePipelineCoupling component.  I need to rename it as the new component does more than just create a new coupling.

File: src/components/Pipelines/CreateOrUpdatePipelineCoupling.jsx [hunk 8]
```diff
@@ -0,0 +1,210 @@
...
+  handleOk = () => {
+    if(!this.props.coupling && (this.state.search === '' || !this.state.search)) {
+      return this.setState({appErrorText:"Select an application to continue"})
+    }
+    this.props.onCreateOrUpdate(this.props.pipeline, this.props.coupling, this.props.stage, this.state.search, this.state.selected);
+  }
+
+  renderLoading = () => (<LinearProgress />);
+
+  renderPipelineDiagram() {
```
Comment (sbeck14) [inline src/components/Pipelines/CreateOrUpdatePipelineCoupling.jsx]: this thing is a work of art

File: src/components/Pipelines/CreateOrUpdatePipelineCoupling.jsx [hunk 9]
```diff
@@ -0,0 +1,210 @@
...
+            <Button className={`${this.props.stage}-cancel`} onClick={this.props.onCancel} color="secondary">Cancel</Button>
+            <Button className={`${this.props.stage}-ok`} onClick={this.handleOk} autoFocus color="primary">Ok</Button>
+          </DialogActions>
+        </Dialog>
+      )
+  }
+}
+
+
+CreatePipelineCoupling.propTypes = {
```
Comment (sbeck14) [inline src/components/Pipelines/CreateOrUpdatePipelineCoupling.jsx]: not a big deal but normally you should also include default props when you specify optional props like:
```
CreatePipelineCoupling.defaultProps = {
coupling: {},
};
```
but I think you're checking for if coupling is null everywhere you use it so we should be OK

Comment (trevorlinton) [reply src/components/Pipelines/CreateOrUpdatePipelineCoupling.jsx]: Yeah this is optional as the component handles both creating a new coupling or updating an existing, so if the coupling is included it assumes its to modify the coupling, if its null it assumes its to create a new coupling.

File: src/components/Pipelines/PipelinePromote.jsx [hunk 10]
```diff
@@ -0,0 +1,359 @@
...
+
+  renderSource() {
+    let description = `${this.props.source.release.build.commit.message || this.props.source.release.description || ''} ${this.props.source.release.build.author || ''}`;
+    let commitUrl = this.props.source.slug.source_blob.url || this.props.source.slug.source_blob.version || '';
+    let commitSha = this.props.source.release.build.commit.sha.substring(0, 7);
+    return (
+      <Paper style={{marginBottom:'0.5rem', ...GlobalStyles.StandardPadding}}>
+        <div style={couplingCardStyle}>
+          <DeveloperBoard style={{marginRight:'0.25rem', ...GlobalStyles.FairlySubtle}}/>
+          <Link style={{...GlobalStyles.Subtle}} style={{fontSize:'16px'}} href={`/apps/${this.state.sourceApp.name}/info`}>{this.state.sourceApp.name}</Link>
```
[the remark this answers is not in the document]
Comment (trevorlinton) [reply src/components/Pipelines/PipelinePromote.jsx]: Added.

File: src/components/Pipelines/PipelinePromote.jsx [hunk 11]
```diff
@@ -0,0 +1,359 @@
...
+      }
+      if(description === '' && coupling.releases.length > 0) {
+        description = coupling.releases[coupling.releases.length - 1].description;
+      }
+      let commitUrl = coupling.slug.source_blob.url || coupling.slug.source_blob.version;
+      return (
+        <Paper key={`target-${coupling.id}`} style={{marginBottom:'0.5rem', ...GlobalStyles.StandardPadding}}>
+          <div style={couplingCardStyle}>
+            <DeveloperBoard style={{marginRight:'0.25rem', ...GlobalStyles.FairlySubtle}}/>
+            <Link style={{...GlobalStyles.Subtle}} style={{fontSize:'16px'}} href={`/apps/${coupling.app.name}/info`}>{coupling.app.name}</Link>
```
[the remark this answers is not in the document]
Comment (trevorlinton) [reply src/components/Pipelines/PipelinePromote.jsx]: done

File: src/components/Pipelines/CreateOrUpdatePipelineCoupling.jsx (hunk 8 above)
Comment (trevorlinton) [reply src/components/Pipelines/CreateOrUpdatePipelineCoupling.jsx]: Oh why thank you. You should see my SVG skills.

**Pull request merged**